multiversx/mx-nft-service
# Development to main  (#984)

* SERVICES1086 rebuild marketplace data/state endpoint, service and functions (part 1)

* SERVICES-1085 extract confusing logic in a separete function; renaming; remove comment

* SERVICES-1085 code splitting, renamings & improvements

* SERVICES-1085 rename param

* SERVICES-1085 add marketplace reindex process event logic

* SERVICES-1085 fix batch slice function; code improvements for get event category function

* SERVICES-1085 remove console log

* SERVICES-1085 extract total bought tokens in a separate function

* SERVICES-1085 split reindex marketplace data code

* SERVICES-1085 use handlers and summary classes for rebuilding marketplace state

* SERVICES-1085 fix marketplace elastic query time range filter

* SERVICES-1085 create MarketplaceReindexState class with common methods and remove common handlers

* SERVICES-1085 add handlers for XOXNO specfic events

* set default paymentNonce if not already set

* SERVICES-1085 ignore not used marketplace events for better performance\

* undo default paymentNonce

* SERVICES-1085 marketplace events summaries

* SERVICES-1085 update marketplace event summaries and add DeadRare updatePrice topics

* SERVICES-1085 marketplace events summary service

* SERVICES-1085 fix marketplace reindex summaries and handle krogan acceptOffer

* SERVICES-1085 marketplace reindex state handlers

* SERVICES-1085 remove confusing method

* SERVICES-1085 remove confusing method

* SERVICES-1085 extract create new order from event summary logic into a separate method

* SERVICES-1085 improve createOrder from summary function

* SERVICES-1085 renamings

* SERVICES-1085 renamings

* SERVICES-1085 update marketplace reindex state service

* SERVICES-1085 process event even if no payment token found

* SERVICES-1085 xoxno user deposit

* SERVICES-1085 fix elrondnftswap offerId type

* SERVICES-1085 reindex marketplace state before & after time range

* SERVICES-1085 reindex marketplace state logs & fixes

* SERVICES-1085 renaming

* SERVICES-1085 fix duplicate bid for internal marketplace

* SERVICES-1085 try-catch and better logging

* SERVICES-1085 fix xoxno no startTime

* SERVICES-1085 fix xoxno no startTime

* SERVICES-1085 fix endAuction with no winning order

* SERVICES-1085 fix case when undefined beforeTimestamp

* SERVICES-1085 save auction tags

* SERVICES-1085 admin endpoints fire and forget

* SERVICES-1085 fix internal auction timestamp topic

* SERVICES-1085 marketplace reindex events locker & logs

* SERVICES-1085 handle multiple internal marketplaces with same address case

* fix admin resolver catch

* SERVICES-1085 exclude collection filtering when unique marketplace address

* SERVICES-1085 fix case when timestamp not spported by DB

* Add plugins module

* Update scam cronjob

* Update scam service

* Add none scam info

* Update workflow

* Update cronjob time

* Remode delete by pattern where not needed

* Fix scam report

* Remove unused code

* trigger cache invalidation

* Change delete to update

* Remove delete by pattern for asset history

* Remove delete by pattern from cache invalidation

* Clear to none nft scam info

* Update scam update cache invalidation

* Add null check

* Add collections with no volum

* Fix trending order by

* Fix trending return

* Fix order trending

* Move blacklist filter before order

* Update filter collection

* Remove clear by pattern in auctions

* Add Scam Collection setting

* Add cache invalidation

* Remove scam info from api

* clean up code

* Upgrade testing packages

* Remove data api calls

* Update mapping

* Refactor Token model

* Add historical price retrieve

* Add elrond tools back

* Add create nft with multiple files

* Extract duplicated code

* Fix typo

* Decrease caching time

* Decrease caching time

* Add ici handling

* Update key

* Update key

* Fix Scam Mapping

* Fix bug for event processing

* Add tickets type on collections

* Add page consts and remove extra pagination

* Remove extra line

* Add isTicket field on asset

* Update redis handler for is ticket

* Add debuging logging

* Fix tickets assets return

* Remove logging

* Update length check

* Update collection length checking

* Fix undefined error

* Allow nullable for role address

---------
